### PR TITLE
Expose more services

### PR DESCRIPTION
**Pull request by Krzysztof-Cieslak**
So this PR does 3 things:

1. Adds `IsPrivateToFile` to `FSharpSymbolUse` - both for VS and FSAC we've been adding it as an extension, but since it's used in both places, I guess we may add it to the type directly
2. Exposes simplify name analyzer
3. Exposes unused declarations analyzer

In general, this PR will allow us to remove the whole bunch of code in FSAC that was copy-pasted from here.

### CODE REVIEW

File: src/fsharp/service/ServiceAnalysis.fsi [hunk 1]
```diff
@@ -8,4 +8,12 @@ open FSharp.Compiler.Range
 
 module public UnusedOpens =
     /// Get all unused open declarations in a file
-    val getUnusedOpens : checkFileResults: FSharpCheckFileResults * getSourceLineStr: (int -> string) -> Async<range list>
\ No newline at end of file
+    val getUnusedOpens : checkFileResults: FSharpCheckFileResults * getSourceLineStr: (int -> string) -> Async<range list>
+
+module public SimplifyNames = 
+    /// Get all ranges that can be simplified in a file
+    val getUnnecessaryRanges : checkFileResults: FSharpCheckFileResults * getSourceLineStr: (int -> string) * sleep: int option -> Async<(range*string) list>
```
[the remark this answers is not in the document]
Comment (cartermp) [reply src/fsharp/service/ServiceAnalysis.fsi]: The `sleep` parameter here (and its implementation) doesn't feel right for inclusion into FCS. It was added to the F# tools before we just turned the feature off by default, and I think it really should just be removed. The feature is expensive but requiring a parameter like this isn't a good answer. Either the feature is expensive and people are okay with that, or we find a way to split it off into its own process. If that work were done in the future, then we'd need to take a breaking change or keep this dummy parameter in here that does nothing.

In other words, I do not want to see "VFT just sweeping the problem under the rug" kind of stuff in the API proper. If a given editor like VFT wants to throttle calls to this API then they should have their own infrastructure for doing it. In our case, I think I'd just rather remove it altogether and see if lacking that configuration setting is even a problem in the first place.

Given that background I think we should remove it and the corresponding call in the VFT. We can do a separate PR that removes the option entirely from the VFT.

File: src/fsharp/service/ServiceAnalysis.fs [hunk 2]
```diff
@@ -226,4 +226,138 @@ module UnusedOpens =
...
+                                | [] -> return current
+                                | headIdent :: restPlid ->
+                                    let! res = checkFileResults.IsRelativeNameResolvableFromSymbol(posAtStartOfName, current, symbolUse.Symbol)
+                                    if res then return current
+                                    else return! loop restPlid (headIdent :: current)
+                            }
+                        loop (List.rev plid) []
+                    
+                    if sleep.IsNone then
+                        do! Async.Sleep sleep.Value  // be less intrusive, give other work priority most of the time
```
Comment (cartermp) [inline src/fsharp/service/ServiceAnalysis.fs]: See comment about the API, this should be deleted

File: src/fsharp/service/ServiceAnalysis.fs [hunk 3]
```diff
@@ -226,4 +226,138 @@ module UnusedOpens =
             let symbolUses = splitSymbolUses symbolUses
             let openStatements = getOpenStatements checkFileResults.OpenDeclarations
             return! filterOpenStatements symbolUses openStatements
+        }
+ 
+module SimplifyNames = 
+    let getPlidLength (plid: string list) = (plid |> List.sumBy String.length) + plid.Length    
+
+    /// Get all ranges that can be simplified in a file
```
Comment (cartermp) [inline src/fsharp/service/ServiceAnalysis.fs]: Comment isn't necessary since the public API has the documentation

**Comment by Krzysztof-Cieslak** [comment]
Updated according to the feedback

**Comment by cartermp** [comment]
@TIHan please review

**Comment by cartermp** [comment]
let's yeet this puppy into the codebase
